tests/core/data_layer/test_data_rpc.py

```diff
@@ -2328,3 +2333,263 @@ async def test_mirrors(
+
+        diff_res = await data_rpc_api.get_kv_diff(
+            {
+                "id": store_id.hex(),
+                "hash_1": hash1.hex(),
+                "hash_2": hash2.hex(),
+                "page": 2,
+                "max_page_size": 5,
+            }
+        )
+        expected_diff = [{"type": "INSERT", "key": key7.hex(), "value": value7.hex()}]
+        assert diff_res == {**diff_reference, "diff": expected_diff}
+
+        diff_res = await data_rpc_api.get_kv_diff(
+            {
+                "id": store_id.hex(),
+                "hash_1": hash1.hex(),
+                "hash_2": hash2.hex(),
+                "page": 3,
+                "max_page_size": 5,
+            }
+        )
+        assert diff_res == diff_reference
+
+        diff_res = await data_rpc_api.get_kv_diff(
+            {
+                "id": store_id.hex(),
+                "hash_1": hash1.hex(),
+                "hash_2": bytes32([0] * 31 + [1]).hex(),
+                "page": 0,
+                "max_page_size": 10,
+            }
+        )
+        empty_diff_reference = {
+            "total_pages": 1,
+            "total_bytes": 0,
+            "diff": [],
+        }
+        assert diff_res == empty_diff_reference
+
+        diff_res = await data_rpc_api.get_kv_diff(
+            {
+                "id": store_id.hex(),
+                "hash_1": bytes32([0] * 31 + [1]).hex(),
+                "hash_2": hash2.hex(),
+                "page": 0,
+                "max_page_size": 10,
+            }
+        )
+        assert diff_res == empty_diff_reference
+
+        new_value = b"\x02\x02"
+        changelist = [{"action": "upsert", "key": key6.hex(), "value": new_value.hex()}]
+        res = await data_rpc_api.batch_update({"id": store_id.hex(), "changelist": changelist})
+        update_tx_rec3 = res["tx_id"]
+        await farm_block_with_spend(full_node_api, ph, update_tx_rec3, wallet_rpc_api)
+
+        history = await data_rpc_api.get_root_history({"id": store_id.hex()})
+        hash1 = history["root_history"][2]["root_hash"]
+        hash2 = history["root_history"][3]["root_hash"]
+
+        diff_res = await data_rpc_api.get_kv_diff(
+            {
+                "id": store_id.hex(),
+                "hash_1": hash1.hex(),
+                "hash_2": hash2.hex(),
+                "page": 0,
+                "max_page_size": 100,
+            }
+        )
+        diff_reference = {
+            "total_pages": 1,
+            "total_bytes": 8,
+            "diff": [
+                {"type": "DELETE", "key": key6.hex(), "value": value6.hex()},
+                {"type": "INSERT", "key": key6.hex(), "value": new_value.hex()},
+            ],
+        }
+        assert diff_res["total_pages"] == diff_reference["total_pages"]
+        assert diff_res["total_bytes"] == diff_reference["total_bytes"]
+        assert sorted(diff_res["diff"], key=lambda x: x["type"]) == diff_reference["diff"]
```

Comment:
Without fixing the order, the test becomes flaky. I believe it's because we use `set` for `kv_diff` (to be consistent with the other method), and set doesn't guarantee to preserve the ordering the same as `list` does (used for `get_keys` and `get_keys_values`)